docker/Dockerfile.omp_offload

```diff
@@ -0,0 +1,110 @@
+##############################################################
+# This Dockerfile contains the additional NVIDIA compilers, 
+# libraries, and plugins to enable OpenMP offloading and NVIDIA GPU 
+# acceleration of Devito codes.
+#
+# TOFIX: Uncomment line "self.cflags += ['-Xopenmp-target', '-march=sm_37']" in devito/arch/compiler.py and change sm_37 for sm_XX where XX is the compute capability of your GPU
+# BUILD: 
+#   sudo docker build -t devito/ompoffload:1.0 -f docker/Dockerfile.omp_offload .
+# 
+# RUN
```

Comment:
I would also add the premade `devitocodes/devito:gpu-dev` image that we build through CI and potentially add this one to CI.
Same for nvidia dockerfile.

A README with all of these in that docker directory may be good with all the different cases building up.